### PR TITLE
Document Presto SQL Query Flow

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: [700](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/700)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/execution/StageInfo.java [hunk 1]
```diff
@@ -125,6 +130,11 @@ public StageStats getStageStats()
         return subStages;
     }
 
+    /**
+     * Gets the map of nodes in SQL plan tree and respective table to be applied.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/StageInfo.java]:
```suggestion
     * Gets the map of nodes in SQL plan tree and their respective table information.
```

File: presto-main/src/main/java/io/prestosql/execution/StageInfo.java [hunk 2]
```diff
@@ -125,6 +130,11 @@ public StageStats getStageStats()
...
+     *
+     * @return the map of nodes in SQL plan tree and respective table to be applied
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/StageInfo.java]:
```suggestion
     * @return the map of nodes in SQL plan tree and their respective table information
```

File: presto-main/src/main/java/io/prestosql/execution/StageStats.java [hunk 3]
```diff
@@ -210,6 +210,11 @@ public DateTime getSchedulingComplete()
         return schedulingComplete;
     }
 
+    /**
+     * Gets an object with statistics about the splits used by a stage.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/StageStats.java:214]: Do you think we should explain what is a "split" here?

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 4]
```diff
@@ -289,12 +289,23 @@ public long getInputPositions()
         return inputPositions;
     }
 
+    /**
+     * Gets the sum of the square of all input positions.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * Gets the squared sum of all input positions.
```

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 5]
```diff
@@ -289,12 +289,23 @@ public long getInputPositions()
...
+     *
+     * @return the sum of the square of all input positions
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * @return the squared sum of all input positions
```

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 6]
```diff
@@ -289,12 +289,23 @@ public long getInputPositions()
...
+     */
     @JsonProperty
     public double getSumSquaredInputPositions()
     {
         return sumSquaredInputPositions;
     }
 
+    /**
+     * Gets the quantity of times that the method to get output buffer from a operator was called
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * Gets the number of times that the method to get output buffer from an operator was called
```

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 7]
```diff
@@ -289,12 +289,23 @@ public long getInputPositions()
...
+     *
+     * @return the quantity of times that the method to get output buffer
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * @return the number of times that the method to get output buffer
```

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 8]
```diff
@@ -289,12 +289,23 @@ public long getInputPositions()
...
+     * from a operator was called
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * from an operator was called
```

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 9]
```diff
@@ -354,6 +365,11 @@ public long getFinishCalls()
         return finishCalls;
     }
 
+    /**
+     * Gets the time used to process all splits.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * Gets the duration time of all splits processing.
```

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 10]
```diff
@@ -354,6 +365,11 @@ public long getFinishCalls()
...
+     *
+     * @return the time used to process all splits
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]:
```suggestion
     * @return the duration time of all splits processing
```

File: presto-main/src/main/java/io/prestosql/operator/PipelineStats.java [hunk 11]
```diff
@@ -314,6 +314,11 @@ public DistributionSnapshot getElapsedTime()
         return elapsedTime;
     }
 
+    /**
+     * Gets the total time that drivers waiting when executing I/O tasks.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/PipelineStats.java]:
```suggestion
     * Gets the total time that drivers waited when executing I/O tasks.
```

File: presto-main/src/main/java/io/prestosql/operator/PipelineStats.java [hunk 12]
```diff
@@ -314,6 +314,11 @@ public DistributionSnapshot getElapsedTime()
...
+     *
+     * @return the total time that drivers waiting when executing I/O tasks
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/operator/PipelineStats.java]:
```suggestion
     * @return the total time that drivers waited when executing I/O tasks
```

File: presto-main/src/main/java/io/prestosql/sql/planner/SystemPartitioningHandle.java [hunk 13]
```diff
@@ -81,6 +81,14 @@ public SystemPartitioningHandle(
         this.function = requireNonNull(function, "function is null");
     }
 
+    /**
+     * Gets the type of used partition.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/SystemPartitioningHandle.java]:
```suggestion
     * Gets the type of the used partition.
```

File: presto-main/src/main/java/io/prestosql/sql/planner/SystemPartitioningHandle.java [hunk 14]
```diff
@@ -81,6 +81,14 @@ public SystemPartitioningHandle(
...
+     * <p>
+     * The type of partition shows if the fragment will be executed only
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/SystemPartitioningHandle.java]:
```suggestion
     * The type of the partition shows if the fragment will be executed only
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanNodeStatsSummarizer.java [hunk 15]
```diff
@@ -62,6 +62,12 @@ private PlanNodeStatsSummarizer() {}
         return aggregatedStats;
     }
 
+    /**
+     * Gets the metadata about the process of node in query plan.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanNodeStatsSummarizer.java]:
```suggestion
     * Gets the metadata about the node process in the query plan.
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanNodeStatsSummarizer.java [hunk 16]
```diff
@@ -62,6 +62,12 @@ private PlanNodeStatsSummarizer() {}
...
+     *
+     * @param taskStats an object that contains the metadata about a running task
+     * @return the metadata about the process of node in query plan
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanNodeStatsSummarizer.java]:
```suggestion
     * @return the metadata about the node process in the query plan
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/TextRenderer.java [hunk 17]
```diff
@@ -373,6 +386,19 @@ private Indent(String firstLinePrefix, String nextLinesPrefix, boolean hasChildr
             this.hasChildren = hasChildren;
         }
 
+        /**
+         * Gets the indentation level for a node in the SQL query tree.
+         * <p>
+         * It is used to construct the visual representation of a
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/TextRenderer.java]:
```suggestion
         * Used to construct the visual representation of a
```

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/io/prestosql/execution/StageStats.java (hunk 3 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/execution/StageStats.java:214]: I do not think so, because there is already a description about it in Presto documentation and it is a pre-requisite to study Presto flow.

**Review by jpedroantunes — APPROVED** [review]
